cpp/src/plasma/test/client_tests.cc

```diff
@@ -202,14 +207,28 @@ TEST_F(TestPlasmaStore, DeleteObjectsTest) {
   ARROW_CHECK_OK(client_.Seal(object_id1));
   ARROW_CHECK_OK(client_.Create(object_id2, data_size, metadata, metadata_size, &data));
   ARROW_CHECK_OK(client_.Seal(object_id2));
-  // Objects are in use.
-  result = client_.Delete(std::vector<ObjectID>{object_id1, object_id2});
-  // TODO: Guarantee that the in-use object will be deleted when it is released.
-  ARROW_CHECK_OK(result);
-  // Avoid race condition of Plasma Manager waiting for notification.
+  // Release the ref count of Create function.
   ARROW_CHECK_OK(client_.Release(object_id1));
   ARROW_CHECK_OK(client_.Release(object_id2));
-  ARROW_CHECK_OK(client_.Delete(std::vector<ObjectID>{object_id1, object_id2}));
+  // Increase the ref count by calling Get.
+  std::vector<ObjectBuffer> object_buffers;
+  ARROW_CHECK_OK(client_.Get({object_id1, object_id2}, 0, &object_buffers));
+  // Objects are still in use.
+  result = client_.Delete(std::vector<ObjectID>{object_id1, object_id2});
+  ARROW_CHECK_OK(result);
+  // The object is occupies and it should not be deleted right now.
+  bool has_object = false;
+  ARROW_CHECK_OK(client_.Contains(object_id1, &has_object));
+  ASSERT_TRUE(has_object);
+  ARROW_CHECK_OK(client_.Contains(object_id2, &has_object));
+  ASSERT_TRUE(has_object);
+  // Decrease the ref count by deleting the ObjectBuffer.
+  object_buffers.clear();
+  // The two objects will not be deleted since Get function increased the ref count.
```

Comment:
Do you mean objects are now deleted in the comment here?

Reply:
Yes, when PlasmaBuffer (in ObjectBuffer) is destructed, it will call Release function. And in Release function, if the ref count is down to 0, it will call Delete function.

Reply:
Ok, so maybe "After decreasing the ref count, the objects are now deleted" is a better comment then?

Reply:
Yes, of course.